impl signer now includes deref impl signer

cavemanloverboy opened the pull request:
Now that `Signer` is implemented for all `Deref<Target = impl Signer>` within `solana-sdk`, we can remove this from the client without introducing breaking changes. This simplifies the code and allows `rust-analyzer` to better infer types.

acheroncrypto reviewed (commented):
> Now that `Signer` is implemented for all `Deref<Target = impl Signer>` within `solana-sdk`, we can remove this from the client without introducing breaking changes.

Unfortunately not because https://github.com/solana-labs/solana/pull/32181 requires `^1.17` but we still support `1.16`.

Also, [nonblocking](https://github.com/coral-xyz/anchor/blob/d28414efd45f726f39fe0f1e19af368357aa6569/client/src/nonblocking.rs) client needs to be updated too.

cavemanloverboy commented:
> > Now that `Signer` is implemented for all `Deref<Target = impl Signer>` within `solana-sdk`, we can remove this from the client without introducing breaking changes.
> 
> Unfortunately not because [solana-labs/solana#32181](https://github.com/solana-labs/solana/pull/32181) requires `^1.17` but we still support `1.16`.
> 
> Also, [nonblocking](https://github.com/coral-xyz/anchor/blob/d28414efd45f726f39fe0f1e19af368357aa6569/client/src/nonblocking.rs) client needs to be updated too.

updated `nonblocking` client, and removed unused `Deref` import. I guess we can wait until 1.17 ships?

acheroncrypto commented:
> updated `nonblocking` client, and removed unused `Deref` import. I guess we can wait until 1.17 ships?

Yes, I think it's better to wait until `1.17` adoption before we make it the minimum version.

cavemanloverboy commented:
> > updated `nonblocking` client, and removed unused `Deref` import. I guess we can wait until 1.17 ships?
> 
> 
> 
> Yes, I think it's better to wait until `1.17` adoption before we make it the minimum version.

🫡

cavemanloverboy commented:
this may be ready to go soon

cavemanloverboy commented:
bump

acheroncrypto commented on client/src/lib.rs:
```diff
@@ -98,35 +97,35 @@ impl<C: Clone + Deref<Target = impl Signer>> Client<C> {
+//         self.0.try_pubkey()
+//     }
 
-    fn sign_message(&self, message: &[u8]) -> solana_sdk::signature::Signature {
-        self.0.sign_message(message)
-    }
+//     fn sign_message(&self, message: &[u8]) -> solana_sdk::signature::Signature {
+//         self.0.sign_message(message)
+//     }
 
-    fn try_sign_message(
-        &self,
-        message: &[u8],
-    ) -> Result<solana_sdk::signature::Signature, solana_sdk::signer::SignerError> {
-        self.0.try_sign_message(message)
-    }
+//     fn try_sign_message(
+//         &self,
+//         message: &[u8],
+//     ) -> Result<solana_sdk::signature::Signature, solana_sdk::signer::SignerError> {
+//         self.0.try_sign_message(message)
+//     }
 
-    fn is_interactive(&self) -> bool {
-        self.0.is_interactive()
-    }
-}
+//     fn is_interactive(&self) -> bool {
+//         self.0.is_interactive()
+//     }
+// }
```
I'm guessing `DynSigner` is no longer required? If so, we should remove it instead of commenting it out.

cavemanloverboy replied:
agreed. i'll take a look at this again
